templates/experiments/components/experiment_version_details_button.html

```diff
@@ -0,0 +1,52 @@
+<button class="btn btn-sm btn-outline btn-primary"
+        onclick="document.getElementById('modal-{{ record.id }}').showModal()">
+    Details
+</button>
+
+<dialog id="modal-{{ record.id }}" class="modal">
```

Comment:
What about using HTMX to fetch the modal content rather than including it all in the main page (https://htmx.org/examples/modal-bootstrap/)

The benefit is a smaller page load for the main page which should be quicker to load and render. We also don't then need to prefetch all the related objects for each version on the main page.